core/src/main/scala/org/apache/spark/storage/BlockManagerStorageEndpoint.scala

```diff
@@ -58,7 +58,9 @@ class BlockManagerStorageEndpoint(
     case RemoveShuffle(shuffleId) =>
       doAsync[Boolean](log"removing shuffle ${MDC(SHUFFLE_ID, shuffleId)}", context) {
         if (mapOutputTracker != null) {
-          mapOutputTracker.unregisterShuffle(shuffleId)
+          // SPARK-53898: call `clearShuffleStatusCache()` explicitly to avoid mistakenly
+          // clearing the single-source-of-truth shuffle status via `unregisterShuffle()`.
```

Comment:
let's also explain the reason: who sends the `RemoveShuffle` request and for what reason.